pkg/data/management/bootstrap_password.go

```diff
@@ -0,0 +1,57 @@
+package management
+
+import (
+	"context"
+	"os"
+
+	"github.com/rancher/wrangler/pkg/randomtoken"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const (
+	bootstrapPasswordSecretName = "bootstrap-secret"
+	bootstrapPasswordSecretKey  = "bootstrapPassword"
+)
+
+// GetBootstrapPassword reads the bootstrap password from it's secret.  If the secret is not found
+// it will be set from `CATTLE_BOOTSTRAP_PASSWORD` or generated if this is empty as well.
+func GetBootstrapPassword(ctx context.Context, lister corev1.SecretInterface) (string, bool, error) {
```

Comment:
A `SecretLister` is a lister, `SecretInterface` is not, it's doing creates and updates.